sdk/core/azure-core-serializer-json-gson/src/main/java/com/azure/core/serializer/json/gson/GsonJsonSerializerProvider.java

```diff
@@ -3,15 +3,15 @@
 
 package com.azure.core.serializer.json.gson;
 
-import com.azure.core.experimental.serializer.JsonSerializer;
 import com.azure.core.experimental.serializer.JsonSerializerProvider;
+import com.azure.core.experimental.serializer.PropertyNameSerializerProvider;
 
 /**
  * Implementation of {@link JsonSerializerProvider}.
  */
-public class GsonJsonSerializerProvider implements JsonSerializerProvider {
+public class GsonJsonSerializerProvider implements JsonSerializerProvider, PropertyNameSerializerProvider {
```

Comment:
I'm wondering if the SPI providers should be different but return the same type, basically copying this class but only having it implement `PropertyNameSerializerProvider`.